programs/keeper/Keeper.cpp

```diff
@@ -623,6 +624,26 @@ try
     buildLoggers(config(), logger());
     main_config_reloader->start();
 
+    std::optional<CgroupsMemoryUsageObserver> observer;
+    try
+    {
+        auto wait_time = config().getUInt64("keeper_server.cgroups_memory_observer_wait_time", 15);
+        if (wait_time != 0)
+        {
+            observer.emplace(std::chrono::seconds(wait_time));
+            observer->setOnMemoryAmountAvailableChanged([&]()
```

Comment:
For Keeper, we don't set memory usage limits (`setMemoryUsageLimits()`), meaning that they will forever remain 0 and we'll do nothing when soft/hard limits are exceeded (CgroupsMemoryUsageObserver.cpp, l. 307). I guess that is okay (not sure)?

What is then the point of setting the callback in setOnMemoryAmountAvailableChanged() then? It will do nothing for Keeper.

Reply:
Actually keeper also has soft limit and will refuse all the requests when the soft limit is met. https://github.com/ClickHouse/ClickHouse/pull/57271

Reply:
Okay then. We should probably add a comment to l. 633 that we are *not* calling `setMemoryUsageLimits()` because soft limits are set otherwise.